esrally/rally.py

```diff
@@ -730,30 +704,183 @@ def generate(cfg):
             mechanic.stop(cfg)
         elif sub_command == "race":
-            race(cfg)
+            # As the race command is doing more work than necessary at the moment, we duplicate several parameters
+            # in this section that actually belong to dedicated subcommands (like install, start or stop). Over time
+            # these duplicated parameters will vanish as we move towards dedicated subcommands and use "race" only
+            # to run the actual benchmark (i.e. generating load).
+            if args.effective_start_date:
+                cfg.add(config.Scope.applicationOverride, "system", "time.start", args.effective_start_date)
+            cfg.add(config.Scope.applicationOverride, "system", "race.id", args.race_id)
+            # use the race id implicitly also as the install id.
+            cfg.add(config.Scope.applicationOverride, "system", "install.id", args.race_id)
+            cfg.add(config.Scope.applicationOverride, "race", "pipeline", args.pipeline)
+            cfg.add(config.Scope.applicationOverride, "race", "user.tag", args.user_tag)
+            cfg.add(config.Scope.applicationOverride, "driver", "profiling", args.enable_driver_profiling)
+            cfg.add(config.Scope.applicationOverride, "driver", "assertions", args.enable_assertions)
+            cfg.add(config.Scope.applicationOverride, "driver", "on.error", args.on_error)
+            cfg.add(config.Scope.applicationOverride, "driver", "load_driver_hosts", opts.csv_to_list(args.load_driver_hosts))
+            cfg.add(config.Scope.applicationOverride, "track", "test.mode.enabled", args.test_mode)
+            configure_track_params(arg_parser, args, cfg)
+            configure_connection_params(arg_parser, args, cfg)
+            configure_telemetry_params(args, cfg)
+            configure_mechanic_params(args, cfg)
+            cfg.add(config.Scope.applicationOverride, "mechanic", "runtime.jdk", args.runtime_jdk)
+            cfg.add(config.Scope.applicationOverride, "mechanic", "source.revision", args.revision)
+            cfg.add(config.Scope.applicationOverride, "mechanic", "car.plugins", opts.csv_to_list(args.elasticsearch_plugins))
+            cfg.add(config.Scope.applicationOverride, "mechanic", "plugin.params", opts.to_dict(args.plugin_params))
+            cfg.add(config.Scope.applicationOverride, "mechanic", "preserve.install", convert.to_bool(args.preserve_install))
+            cfg.add(config.Scope.applicationOverride, "mechanic", "skip.rest.api.check", convert.to_bool(args.skip_rest_api_check))
+
+            configure_reporting_params(args, cfg)
+
+            race(cfg, args.kill_running_processes)
         elif sub_command == "generate":
+            cfg.add(config.Scope.applicationOverride, "generator", "chart.type", args.chart_type)
+            cfg.add(config.Scope.applicationOverride, "generator", "output.path", args.output_path)
+            configure_track_params(arg_parser, args, cfg)
```

Comment:
This is blocking generation.

Reply:
Fixed in https://github.com/elastic/rally/pull/1175/commits/c89aaaef84e521a6049561da50eb032ea74d6116.